down_frab_videos.py

```diff
@@ -241,6 +243,28 @@ def __init__(self, short_message, long_message):
         self.long_message = long_message
 
 
+def wrap_bs4(content):
+    try:
+        return BeautifulSoup(content, "lxml")
+    except bs4.FeatureNotFound:
+        print("Warning: could not parse with 'lxml', check your installation. "
+              "Falling back to 'html.parser'.")
+
+    try:
+        return BeautifulSoup(content, "html.parser")
+    except bs4.FeatureNotFound:
+        print("Warning: could not parse with 'html.parser', check your installation. "
+              "Falling back to 'html5lib'.")
+
+    try:
+        return BeautifulSoup(content, "html5lib")
+    except bs4.FeatureNotFound:
+        print("Error: could not parse with any of 'lxml', 'html.parser' and 'html5lib'. "
+              "Giving up.")
```

Comment:
From the BeautifulSoup docs, I would say that `html.parser` is the worst and should be used last, right?

Reply:
Depends on your figure of merit. But let's put `html.parser` last.